styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```
@@ -210,6 +216,21 @@ public void close() throws IOException {
     }
   }
 
+  @Override
+  public void restore() {
+    final Retryer<Object> retryer = RetryerBuilder.newBuilder()
+        .retryIfException()
+        .withWaitStrategy(WaitStrategies.exponentialWait())
+        .withStopStrategy(StopStrategies.stopAfterAttempt(3))
+        .build();
+
+    try {
+      retryer.call(Executors.callable(this::tryPollPods));
+    } catch (ExecutionException | RetryException e) {
+      throw new RuntimeException(e);
```

Why is the retry logic needed at all? It's part of a startup routine, I think it's ok to fail hard here.

Is it expected to fail intermittently?